dask/dataframe/partitionquantiles.py

```diff
@@ -419,7 +425,14 @@ def percentiles_summary(df, num_old, num_new, upsample, state):
     if is_categorical_dtype(data):
         data = data.cat.codes
         interpolation = "nearest"
-    elif is_datetime64_dtype(data.dtype) or is_integer_dtype(data.dtype):
+    elif (
+        is_datetime64_dtype(data.dtype)
+        or is_integer_dtype(data.dtype)
+        or (
+            hasattr(data.dtype, "numpy_dtype")
+            and is_integer_dtype(data.dtype.numpy_dtype)
+        )
```

Comment:
Similar comment about moving this into an `is_integer_dtype` compatibility function